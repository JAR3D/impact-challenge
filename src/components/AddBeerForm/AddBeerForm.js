import { useState } from "react";

const AddBeerForm = ({ addBeer }) => {
  const [formData, setFormData] = useState({});

  const handleSubmit = (e) => {
    e.preventDefault();
    addBeer({
      nameDisplay: formData.name,
      description: formData.description,
      labels: { medium: formData["label-url"] },
    });
  };

  const onChange = (e) => {
    setFormData({
      ...formData,
      [e.currentTarget.name]: e.currentTarget.value,
    });
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="name">Name:</label>
      <input
        type="text"
        name="name"
        id="name"
        value={formData.name}
        onChange={onChange}
      />

      <label htmlFor="description">Description:</label>
      <input
        type="text"
        name="description"
        id="description"
        value={formData.description}
        onChange={onChange}
      />

      <label htmlFor="label-url">Label url:</label>
      <input
        type="text"
        name="label-url"
        id="label-url"
        value={formData["label-url"]}
        onChange={onChange}
      />

      <button type="submit">Add Beer</button>
    </form>
  );
};

export default AddBeerForm;
