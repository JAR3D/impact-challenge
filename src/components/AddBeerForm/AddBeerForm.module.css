.form {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.form > div {
  display: flex;
  justify-content: space-between;
  margin-bottom: 5px;
  width: 250px;
}

.form > button {
  width: 250px;
}
