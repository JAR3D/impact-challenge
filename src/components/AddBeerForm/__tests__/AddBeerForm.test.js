import { render, cleanup, screen, waitFor } from "@testing-library/react";
import user from "@testing-library/user-event";

import AddBeerForm from "../AddBeerForm";

describe("AddBeerForm", () => {
  beforeEach(() => cleanup());

  it("should render", () => {
    const { baseElement } = render(<AddBeerForm />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should submit", async () => {
    const mockAddBeer = jest.fn();
    render(<AddBeerForm addBeer={mockAddBeer} />);
    await user.type(screen.getByLabelText(/name/i), "Foo");
    await user.type(screen.getByLabelText(/description/i), "Bar");
    await user.type(screen.getByLabelText(/label/i), "some url");
    await user.click(screen.getByText(/add beer/i));

    await waitFor(() => expect(mockAddBeer).toHaveBeenCalledTimes(1));
  });
});
