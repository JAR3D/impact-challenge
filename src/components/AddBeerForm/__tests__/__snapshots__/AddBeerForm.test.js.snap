// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBeerForm should render 1`] = `
<body>
  <div>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value=""
      />
      <label
        for="description"
      >
        Description:
      </label>
      <input
        id="description"
        name="description"
        type="text"
        value=""
      />
      <label
        for="label-url"
      >
        Label url:
      </label>
      <input
        id="label-url"
        name="label-url"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Beer
      </button>
    </form>
  </div>
</body>
`;
